back/app/policies/project_policy.rb

```diff
@@ -4,29 +4,38 @@ class Scope
 
     def initialize(user, scope)
       @user  = user
-      @scope = scope
+      @scope = scope.includes(:admin_publication)
     end
 
     def resolve
-      if user&.admin?
-        scope.all
-      elsif user&.project_moderator?
-        scope.where(id: user.moderatable_project_ids + scope.user_groups_visible(user).or(scope.publicly_visible).ids)
-      elsif user
-        scope.user_groups_visible(user).not_draft.or(scope.publicly_visible.not_draft)
-      else
-        scope.publicly_visible.not_draft
-      end
+      # Resolves the scope as a disjunction (OR) of scopes, one scope (= one clause) for each 'role' a user can have.
+      # It entails that scopes does not have to be redundant. In other words, each sub-scope (clause) should aim to
+      # include only the projects to which this role gives access (without repeating projects to which lesser roles
+      # of the user gives access).
+      resolve_for_admin
+        .or resolve_for_visitor
+        .or resolve_for_normal_user
```

Comment:
This pattern is reused in a few policies. Then, extensions add new clauses to the disjunction (= new ORs), or redefine existing clauses.

Reply:
Damn it! I almost refactored that part in another lifetime (despite the blame timestamps, long story...). I think it's a refactoring from a previous version of the scope. See: https://github.com/CitizenLabDotCo/citizenlab/commit/4c2153b57cccb0a38a89270cb808b01b2adfafcf

Now, the only difference I see concerns moderators. It seems they have access to draft projects now and I really don't understand why.

Reply:
Hi @adessy at first glance it does not sound like expected behavior. Could you point me to the change that made you believe that?

Reply:
It's just here in the commented diff 🙂 
```ruby
	  elsif user&.project_moderator?
	    scope.where(id: user.moderatable_project_ids + scope.user_groups_visible(user).or(scope.publicly_visible).ids)
	  elsif user
	    scope.user_groups_visible(user).not_draft.or(scope.publicly_visible.not_draft)
```
the scope is filtered with `.not_draft` in the user clause, but not in the moderator clause.

Reply:
- `scope.user_groups_visible(user).not_draft.or(scope.publicly_visible.not_draft)` for users
- `scope.user_groups_visible(user)._________.or(scope.publicly_visible._________)` + moderated projects for moderators

Reply:
Not sure if my implementation is correct anymore, but the reasoning was that users should never be able to see draft projects, while project managers should be able to preview projects in any state in the home or folder page.